ls1intum/Artemis
# [Feature] - Course Management Detail View

### Checklist

### Motivation and Context


The course management detail view should be integrated more deeply into the workflow of the instructors and tutors. Currently the detail page is not used very frequently since it only stores basic course information which is not useful for daily tasks

### Description

In the first iteration of this page, we add an active user chart and some doughnut charts to visualize some statistics regarding the course

### Steps for Testing


1. Log in to Artemis
2. Navigate into Course Management Overview
3. Click on Course header
4. In the Course Management Detail View, make sure that no weird numbers show up which are not rounded for example
5. Create a new exercise and participate with min. 2 users
6. Check in the detail view that two more assessments need to be done
7. Assess those and create a complaint with the one and a more feedback request with the other one
8. Check in the detail view that the corresponding numbers in the doughnut charts go up
9. Check that the active users rise by the amount of user with which you made submissions
10. Check that the average score of the course is correct (average student score is shown in the course statistics and in the participant score)

### Test Coverage



course-detail.component.ts: 93%
course-detail-bar-chart.component.ts: 95%
course-detail-doughnut-chart.component.ts: 89%
CourseResource.java: 93%

### Screenshots


Example of a course detail page
![image](https://user-images.githubusercontent.com/44401560/116396421-a7f46480-a825-11eb-9d8b-14e018e32df3.png)

We also still provide the old course data below the charts:
<img width="1436" alt="Screenshot 2021-05-04 at 13 48 20" src="https://user-images.githubusercontent.com/44401560/116999345-c2c05080-acdf-11eb-8226-1d9b416a6d27.png">

In comparison, before, it looked like this:
![Screenshot 2021-05-04 at 13 51 52](https://user-images.githubusercontent.com/44401560/116999473-f4391c00-acdf-11eb-80f7-b06b4d277e61.png)
